### ISSUE TITLE
Feature Request: Also track keyboard only form changes for `select` (affects Userfields)

### ISSUE THREAD

**Issue by josephshanak** [grocy/grocy#2901]
Re-submitting since my former issue (#2900) was closed, locked, and changed to a feature request (but left closed without a reason why this does not belong in Grocy). I think this is an accessibility issue, and frustrating for users who prefer using the keyboard to input forms, but if you think it should be tracked as a feature request that is fine. 

NOTE: None of this issue, or the prev issue included *any* AI generated text or involved the the use of any AI tools whatsoever.

Please let me know if you need any more information.

# Feature Request

I want to be able use the keyboard to enter user fields. (e.g. [TAB] to move between fields, arrow to select items in a `select`. Right now this does not appear to be possible.

## What happens currently

* Add a UserField
   * Choose Select List (Single)
   * Choose "Show as columns in table" (not required but easier to notice issue)

<img width="648" height="943" alt="Image" src="https://github.com/user-attachments/assets/c8235251-81e0-408f-b725-421adb3e3e79" />

* Create an entity with that userfield
   * IMPORTANT: Use only your keyboard for navigation
   * "Custom Field 1" [TAB] "Custom Field 2" [TAB] [ARROW DOWN] [ARROW DOWN] [ENTER] [TAB]

<img width="648" height="381" alt="Image" src="https://github.com/user-attachments/assets/3c9e7f52-77a5-49ab-8985-595473d6b771" />
 
* User field is not saved:

<img width="3211" height="999" alt="Image" src="https://github.com/user-attachments/assets/7af29238-6f72-43e5-a021-0da54934f056" />

It's also worth noting that if the field is marked as `mandatory`, it is still saved with a blank value.

## Debugging

After you select the userfield using your keyboard, if you inspect the form you will see that the `is-dirty` class is not added to the `select`:

<img width="946" height="505" alt="Image" src="https://github.com/user-attachments/assets/c54bd0bc-4be9-4d21-993c-c5259db03a7b" />

I traced this back to this event handler, which does not appear to be called when following the workflow above:

https://github.com/grocy/grocy/blob/c0c9a38e396aaf437d923f288c3325b54f679a7d/public/js/grocy.js#L502

# Proposed Solution

Since you disagreed with my suggestion in #2900, I'll leave the implementation details up to you. 

# Versions

Browser: Firefox 148 (aarch64)
OS: MacOS Sequoia 15.7.4
Grocy: Demo Instance

<img width="1258" height="814" alt="Image" src="https://github.com/user-attachments/assets/e2563ec4-30d3-4752-a0db-8899bc71b307" />
